pkg/cspc/algorithm/select_node.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package algorithm
+
+import (
+	"github.com/openebs/maya/pkg/volume"
+	"github.com/pkg/errors"
+	openebsio "github.com/openebs/api/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/api/pkg/apis/types"
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+// SelectNode returns a node where pool should be created.
+func (ac *Config) SelectNode() (*cstor.PoolSpec, string, error) {
+	usedNodes, err := ac.GetUsedNode()
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "could not get used nodes list for pool creation")
+	}
+	for _, pool := range ac.CSPC.Spec.Pools {
+		// pin it
+		pool := pool
+		nodeName, err := ac.GetNodeFromLabelSelector(pool.NodeSelector)
+		if err != nil || nodeName == "" {
+			klog.Errorf("could not use node for selectors {%v}", pool.NodeSelector)
```

Comment:
Let us log cspc name for better understanding of logs

Reply:
final logger logs the cspc name